lesson-05-time-management/xxx_timer.c

```diff
@@ -0,0 +1,105 @@
+#include <linux/module.h>
+#include <linux/fs.h>
+#include <linux/cdev.h>
+#include <asm/uaccess.h>
+#include <linux/pci.h>
+#include <linux/version.h>
+#include <linux/init.h>
+#include <linux/time.h>
+#include <linux/timer.h>
+
+
+#define LEN_MSG 160
+#define DELAY HZ
+static unsigned int fib_n;
+static char* buf_msg;
+static struct timer_list irq_timer;
+//static unsigned long time_interval;
+
+unsigned long long int fib(unsigned int n)
+{
+   unsigned long long int first = 0, second = 1, next = 0;
+   unsigned int i;
+   for ( i = 0 ; i <= n ; i++ )
+   {
+      if ( i <= 1 )
+         next = i;
+      else
+      {
+         next = first + second;
+         first = second;
+         second = next;
+      }
+   }
+   return next;
+}
+
+static ssize_t xxx_show( struct class *class, struct class_attribute *attr, char *buf ) {
+   
+   strcpy( buf, buf_msg );  
+//   printk("timeout: %ld", time_interval);
+   printk( "read %ld\n", (long)strlen( buf ) );   
+   return strlen( buf );
+}
+
+static ssize_t xxx_store( struct class *class, struct class_attribute *attr, const char *buf, size_t count ) {
+//   printk( "write %ld\n", (long)count );
+   strncpy( buf_msg, buf, count );
+   buf_msg[ count ] = '\0';
+   return count;
+}
+
+CLASS_ATTR_RW( xxx );
+
+static struct class *x_class;
+
+void IRQ_TimerHandler(unsigned long data)
+{
+	char buf[100];
+//	time_interval++;
+	sprintf(buf,"%u %llu\n",fib_n,fib(fib_n));
+//   	printk( KERN_INFO "%s\n", buf );
+   	xxx_store(x_class, &class_attr_xxx,buf,strlen(buf)+1);
+	if (fib_n < 93) fib_n++;
+   	else fib_n = 0;
```

Comment:
Good intention to handle overflow at least somehow,
But I believe, the task assumed a bit more than 93 seconds (at least not via magic number).